Eastern-Research-Group/expert-query
# Feature/73 local docker database

## Related Issues:
* [EQ-73](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=865&view=detail&selectedIssue=EQ-73)

## Main Changes:
* Created Docker image and Docker compose configurations to facilitate local database interactions.
* Included a short script for loading bulk ATTAINS data into one of the containers.

## Steps To Test:
1. The mock OWPUB database can be started by running `docker compose up` or `docker-compose up` while inside the `owpub` directory
2. The database should then be exposed at `localhost:5433`, with an admin interface at `localhost:8081`
3. To log into the admin interface, select **PostgreSQL** from the drop-down menu, then input **owpub**, **postgres**, **example**, **postgres** for the Server, Username, Password, and Database fields, respectively
4. To load sample data into the OWPUB database, run `docker ps`, get the container ID affiliated with the image **expert-query/postgis**, then connect to the container with `docker exec -it [container_id] /bin/bash`
5. Run the `load-data.sh` script with the file location of the current OGC Geopackage, i.e. `/usr/local/bin/load-data.sh https://edap-ow-data-commons.s3.amazonaws.com/data/ATTAINS_Assessment_20220809_gpkg.zip` **WARNING: this file is 11GB when unzipped**
6. The local Postgresql database can be started by running the same `docker compose up` or `docker-compose up` command inside the `postgres` directory
7. This database will then be reachable at `localhost:5432`, with an admin interface at `localhost:8080`
8. The Server name in this case is **postgres**, the other fields are the same as those used for the **owpub** instance
